source/auth/Credentials.cpp

```diff
@@ -106,7 +116,7 @@ namespace Aws
 
             CredentialsProvider::~CredentialsProvider()
             {
-                if (m_provider != nullptr)
+                if (m_provider)
```

Comment:
This is mainly to raise the question:
Do we need to call `aws_credentials_provider_acquire(provider)` as well in the constructor above?
To me it seems we should.